### DISCUSSION TITLE
getSession always returns null.

### DISCUSSION THREAD

**Discussion by SFin12** [prisma/prisma#19276] [Q&A]
I followed the tutorial up until the section where a user create a post.  Within the following code copied directly from the tutorial, I keep getting getting the same error.  When I log "session" after it returns it is always null even though I am logged in and the user info is all stored in the database. The only change I have made was allowing google-sign in rather than just the github sign-in.
```
// pages/api/post/index.ts

import { getSession } from 'next-auth/react';
import prisma from '../../../lib/prisma';

// POST /api/post
// Required fields in body: title
// Optional fields in body: content
export default async function handle(req, res) {
  const { title, content } = req.body;

  const session = await getSession({ req });
console.log(session) // always null even when signed in
  const result = await prisma.post.create({
    data: {
      title: title,
      content: content,
      author: { connect: { email: session?.user?.email } },
    },
  });
  res.json(result);
}
```
Here is the error:

> [next-auth][error][CLIENT_FETCH_ERROR] 
> https://next-auth.js.org/errors#client_fetch_error undefined {
>   error: {},
>   url: 'http://localhost:3000/api/auth/session',
>   message: undefined
> }
> session null

Because of this I can't get anything to post to the database. 

After searching for 3+ hours and trying many alternatives I haven't been able to resolve this.

**Comment by jharrell** [comment]
Hi @SFin12, sorry about the trouble. What tutorial, specifically? I will run through it and see if anything jumps out at me.

**Reply by SFin12** [reply]
Thanks @jharrell , here is the link to the tutorial: https://vercel.com/guides/nextjs-prisma-postgres

**Reply by jharrell** [reply]
There are a few things we can fix up/edit in that guide and I'm also running into issues! 

Just to confirm @SFin12, you're running into trouble around Step 7 and 8, is that right? 

I'll have to dig deeper into this and most likely publish some updates for the guide. Thank you for pointing this out!

**Reply by viktor992** [reply]
I ran into the same issue and I managed to make it work by using the `getServerSession` method in the api route implementation. Here's more information about my implementation: #19400 . If you agree, we could modify the guide to include these changes.

**Comment by SFin12** [comment]
@jharrell I'm mainly stuck on 8 for this problem.  I'm able to sign in ( I added google sign-in along with Github) and I can see the session details on the front-end when I log it by using useSession but I'm having issues using getSession on the backend within the pages/api/post route. It throws a prisma error as well but that's only because my session is always null even if I'm signed in.

**Comment by SFin12** [comment]
To move forward with the tutorial for now, I sent the session data with the post so Prisma can use it.  Probably not a secure way to do but it's a work around for now. What's interesting is that getSession seems to work within getServerSideProps for step 9 but always returns null when used in step 8.

**Reply by viktor992** [reply]
Hey @SFin12 , maybe this could be useful to you as well: #19400

**Reply by SFin12** [reply]
Awesome! I should have a chance to refactor tonight to make sure it works on my side.

**Reply by ludralph** [reply]
Hi @SFin12 

do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
